lib/base/dependencygraph.cpp

```diff
@@ -5,46 +5,63 @@
 using namespace icinga;
 
 std::mutex DependencyGraph::m_Mutex;
-std::map<Object *, std::map<Object *, int> > DependencyGraph::m_Dependencies;
+DependencyGraph::DependencyMap DependencyGraph::m_Dependencies;
 
 void DependencyGraph::AddDependency(Object *parent, Object *child)
 {
 	std::unique_lock<std::mutex> lock(m_Mutex);
-	m_Dependencies[child][parent]++;
+	Edge edge(parent, child, 1);
+	if (auto it(m_Dependencies.find(edge)); it != m_Dependencies.end()) {
+		m_Dependencies.modify(it, [](Edge& e) { e.count++; });
+	} else {
+		m_Dependencies.insert(edge);
```

Comment:
Also I'm wondering whether the new map

* either allows to pass an iterator as hint on insertion
* or supports kind of insert-if-new-else-return-old-position op

Reply:
> * either allows to pass an iterator as hint on insertion

> Returns: The return value is a pair p. p.second is true if and only if insertion took place. On successful insertion, p.first points to the element inserted; otherwise, p.first points to an element that caused the insertion to be banned.